ctapipe/calib/camera/calibrator.py

```diff
@@ -134,6 +138,9 @@ def _calibrate_dl1(self, event, telid):
         if self._check_dl0_empty(waveforms):
             return
 
+        selected_gain_channel = event.r1.tel[telid].selected_gain_channel
+        time_shift = event.calibration.tel[telid].dl1.time_shift
```

Comment:
If I understood it well, there is only one time_shift per pixel and event, which corresponds to the selected gain channel. But for some events (flat-field & pedestals) we will keep both channels. If in such cases the R1 event source fills both waveforms, HG and LG, it should fill also the time_shift values for both. Otherwise, in order to analyze the two gains one would have to use one event source selecting always HG and another event source selecting always LG. May be it is not a big deal, but I think if at some point the data contains both waveforms it should also contain the additional info needed to properly use them.

Reply:
All code currently in ctapipe assumes that dl0 data does not have two gain channels.

If that does not hold true, we need a broader change in ctapipe which should not be part of this PR.

Reply:
You should maybe open an issue about that.

Reply:
I think the double gain for those events will exist in R1, for calibration purposes, but in DL0 the _result_ of those calibrations should already go into the monitoring.  Since you are passing the 'selected_gain' I assumed both were still there, but the non-selected one will already be empty at that point, ok.

Reply:
Ah, makes sense, so probably we are fine.